scte35/segmentationdescriptor.go

```diff
@@ -255,19 +257,20 @@ func (d *segmentationDescriptor) CanClose(out SegmentationDescriptor) bool {
 			return true
 		}
 	case segCloseEventIDNotNested:
+		// this should also consider segnum == segexpected for IN signals closing an out signal.
+		if d.IsIn() && d.EventID() == out.EventID() && d.SegmentNumber() == d.SegmentsExpected() {
+			return true
+		}
 		if d.EventID() != out.EventID() {
 			return false
 		}
 		fallthrough
-	case segCloseNotNested:
-		if d.IsIn() {
-			// if in, last desc is x/x
-			if d.segNum == d.segsExpected {
+	case segCloseNotNested: // only applies to 0x34 and 0x36 with subsegments.
+		if d.IsOut() && (d.TypeID() == SegDescProviderPOStart || d.TypeID() == SegDescDistributorPOStart) {
+			if d.HasSubSegments() && d.SubSegmentNumber() == d.SubSegmentsExpected() {
 				return true
 			}
-		} else if d.IsOut() {
-			// if out, first descriptor in set closes existing open
-			if d.segNum == 1 {
+			if d.TypeID() == out.TypeID() {
```

Comment:
Same story here, I think this is already done in the close rules map and this is just extra, unneeded logic.

Reply:
Agreed.